src/Repositories/TopicRepository.php

```diff
@@ -67,27 +74,63 @@ public function model()
      *
      * @return array list of unique classes
      */
-    public static function registerContentClass(string $class): array
+    public function registerContentClass(string $class): array
+    {
+        if (!in_array($class, $this->contentClasses) && class_exists($class) && (is_a($class, TopicContentContract::class, true))) {
+            $this->contentClasses[] = $class;
+        }
+
+        return $this->contentClasses;
+    }
+
+    public function registerContentClasses(array $classes): array
+    {
+        foreach ($classes as $class) {
+            $this->registerContentClass($class);
+        }
+
+        return $this->contentClasses;
+    }
+
+    public function registerResourceClass(string $topicTypeClass, string $resourceClass, string $type = 'client'): array
+    {
+        if (!isset($this->resourceClasses[$type][$topicTypeClass]) && class_exists($resourceClass) && (is_a($resourceClass, JsonResource::class, true))) {
+            $this->resourceClasses[$type][$topicTypeClass] = $resourceClass;
+        }
+
+        return $this->resourceClasses[$type];
+    }
+
+    public function registerResourceClasses(string $topicTypeClass, array $resourceClasses): array
+    {
+        foreach ($resourceClasses as $type => $resourceClass) {
+            $this->registerResourceClass($topicTypeClass, $resourceClass, $type);
+        }
+
+        return $this->resourceClasses;
+    }
+
+    public function getResourceClass(string $topicTypeClass = null, string $type = 'client'): string
     {
-        if (!in_array($class, self::$contentClasses) && class_exists($class) && (is_a($class, TopicContentContract::class, true))) {
-            self::$contentClasses[] = $class;
+        if (isset($this->resourceClasses[$type][$topicTypeClass])) {
+            return $this->resourceClasses[$type][$topicTypeClass];
         }
 
-        return self::$contentClasses;
+        return false;
```

Comment:
`return false` in method with `string` return type will throw error

and I think it would be better to add a `public function hasResourceClass(string $topicTypeClass, string $type): bool` so that calls could look
```php
if(Topic::hasResourceClass($this->topicable_type, 'client')){ 
$class = Topic::getResourceClass(); 
}
```

or maybe even better would be adding a `public function makeResourceClass(TopicContentContract $topicable, string $type = 'client')` method to this Repository that would return `: ?JsonResource`?